src/completions/Tab.ts

```diff
@@ -139,6 +139,28 @@ export class BufferCompletionSource extends Completions.CompletionSourceFuse {
         }
     }
 
+    /** Return the scoredOption[] result for the tab index startswith n */
+    private TabscoredOptionsStartsWithN(
+        n: number,
+        options: BufferCompletionOption[]
+    ): Completions.ScoredOption[] {
+        const nstr = (n + 1).toString()  // Make sure log(n) is valid
+        const res = [];
+        for (const [index, option] of enumerate(options)) {
+            if ((option.tabIndex + 1).toString().startsWith(nstr)) {
+                res.push({
```

Comment:
Quite possible I'm missing something here, but why add 1 to both nstr and tabindex?

Reply:
because if you input `1` here,  the prefix will be `0`, I think, which match only the `tabIndex = 0`, not 11, 12, 13...

Reply:
```
nstr = n +1 + ""
(option.tabIndex +1 +""). startsWith (nstr)

// Identical to
n2 = n + ""
(option.tabIndex + "").startsWith(n2)
```

where `+""` is a hacky way of converting to strings because I'm on my phone and I wanted to be terse.

What am I missing?

Reply:
Because well, we are not using math.

When `n = 1`, we actually want to match tabs with prefix `2`. That is `tabIndex = 1, 19 20, 21, 22, 23, ...28, 199, 200, ...` which translated to 1 based `2, 20, 21, 22, 23, 24, ...29, 200, 201...`

Now, only `tabIndex = 1, 10, 11, ...19` match `n = 1`. In 1-base number, those translated to tab `2, 11,  12, ... 20`

Reply:
```
(x + k + "").startsWith(n + k + "") == (x + "").startsWith(n + "") // true for all integers x,k,n
```

I think lines 147 and 150 are equivalent to the left hand side with x:=option.tabIndex and n:=n

Reply:
Then, you just have to change the code to see the different yourself.

Reply:
@mozbugbox Sorry, bovine3dom asked me to take a look at this PR and I forgot about it.

@bovine3dom Our Tridactyl tab indexes are not Firefox's. Ours are 1-based and Firefox's are 0-based. So we need to add 1 to n. It's fine to not add 1 on either side of the equation as long as all tab indexes are in [0-9], but once we have a tabIndex equal to or greater than 10, we want `1` to match both `1` and `10`, which isn't something we can do if we keep our 0-based indexes.

Reply:
Ah, I see now. My identity was just wrong but I hadn't spotted that a few lines above we subtract one from the user input, so here we add one back on to it. Thanks both, and sorry for being a bit slow :)